### PR TITLE
DHFPROD-4845: Refine filtered results so nested elements with no filtered results do not appear

### PR DESCRIPTION
**Pull request by xnikhil08**
### Description
This PR implements the feature given in the ticket DHFPROD-4845.

#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

File: marklogic-data-hub-central/ui/src/components/entities/mapping/source-entity-map/source-to-entity-map.test.tsx [hunk 1]
```diff
@@ -44,88 +44,172 @@ describe('RTL Source-to-entity map tests', () => {
...
         />);
 
-      //Expanding all the nested levels first to apply filter later
-      fireEvent.click(getByTestId('expandCollapseBtn-source'));
-      fireEvent.click(getByTestId('expandCollapseBtn-entity'));
+        /* Test filter for JSON Source data in Source table  */
+        let filterIcon = getByTestId('filterIcon-key');
+        expect(filterIcon).toBeInTheDocument();
+        fireEvent.click(filterIcon);
+        let inputSearch = getByTestId('searchInput-key');
```
Comment (bsrikan) [inline marklogic-data-hub-central/ui/src/components/entities/mapping/source-entity-map/source-to-entity-map.test.tsx:59]: lets initialize resetSearch here. all the elements here, to go with arrange/act/assert way of writing tests.

File: marklogic-data-hub-central/ui/src/components/entities/mapping/source-entity-map/source-to-entity-map.test.tsx [hunk 2]
```diff
@@ -44,88 +44,172 @@ describe('RTL Source-to-entity map tests', () => {
...
+            sourceData={data.jsonSourceDataMultipleSiblings}
+            entityTypeProperties={data.entityTypePropertiesMultipleSiblings}
         />);
 
-      //Expanding all the nested levels first to apply filter later
-      fireEvent.click(getByTestId('expandCollapseBtn-source'));
-      fireEvent.click(getByTestId('expandCollapseBtn-entity'));
+        /* Test filter for JSON Source data in Source table  */
+        let filterIcon = getByTestId('filterIcon-key');
+        expect(filterIcon).toBeInTheDocument();
```
Comment (bsrikan) [inline marklogic-data-hub-central/ui/src/components/entities/mapping/source-entity-map/source-to-entity-map.test.tsx:57]: Click will assert that element to be in the dom. We can get rid of the assertion here, line 60, 62 and other places if present.

File: marklogic-data-hub-central/ui/src/components/entities/mapping/source-entity-map/source-to-entity-map.test.tsx [hunk 3]
```diff
@@ -44,88 +44,172 @@ describe('RTL Source-to-entity map tests', () => {
...
+        expect(getByTestId('submitSearch-key')).toBeInTheDocument();
+        expect(inputSearch).toHaveValue('organism');
+        fireEvent.click(getByTestId('submitSearch-key')); //Click on Search button to apply the filter with the desired string
+
+        //Check if the expected values are available/not available in search result.
+        expect(getByText(/withNuts:/)).toBeInTheDocument();
+        expect(getByText('Frog virus 3')).toBeInTheDocument();
+        expect(getByText('scientific')).toBeInTheDocument();
+        expect(getByText(/nutFree:/)).toBeInTheDocument();
+        expect(queryByText('NamePreferred')).not.toBeInTheDocument();
```
Comment (bsrikan) [inline marklogic-data-hub-central/ui/src/components/entities/mapping/source-entity-map/source-to-entity-map.test.tsx:188]: All the "not" here are siblings right?

Comment (xnikhil08) [reply marklogic-data-hub-central/ui/src/components/entities/mapping/source-entity-map/source-to-entity-map.test.tsx:188]: Yes.

File: marklogic-data-hub-central/ui/src/components/entities/mapping/source-entity-map/source-to-entity-map.test.tsx (hunk 2 above)
Comment (xnikhil08) [reply marklogic-data-hub-central/ui/src/components/entities/mapping/source-entity-map/source-to-entity-map.test.tsx]: Done.

File: marklogic-data-hub-central/ui/src/components/entities/mapping/source-entity-map/source-to-entity-map.test.tsx (hunk 1 above)
Comment (xnikhil08) [reply marklogic-data-hub-central/ui/src/components/entities/mapping/source-entity-map/source-to-entity-map.test.tsx]: Done.

**Pull request merged**